scripts/verify-install.sh

```diff
@@ -29,195 +35,289 @@ function check_command() {
   fi
 }
 
-# By default we're picking up lnd and lncli from the system $PATH.
-LND_BIN=$(which lnd)
-LNCLI_BIN=$(which lncli)
+function verify_version() {
+  version_regex="^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]"
+  if [[ ! "$1" =~ $version_regex ]]; then
+    echo "ERROR: Invalid expected version detected: $1"
+    exit 1
+  fi
+  echo "Expected version for binaries: $1"
+}
 
-# If exactly two parameters are specified, we expect the first one to be lnd and
-# the second one to be lncli.
-if [[ $# -eq 2 ]]; then
-  LND_BIN=$(realpath $1)
-  LNCLI_BIN=$(realpath $2)
-  
-  # Make sure both files actually exist.
-  if [[ ! -f $LND_BIN ]]; then
-    echo "ERROR: $LND_BIN not found!"
+function import_keys() {
+  # A trick to get the absolute directory where this script is located, no
+  # matter how or from where it was called. We'll need it to locate the key
+  # files which are located relative to this script.
+  DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+
+  # Import all the signing keys. We'll create a key ring for each user and use
+  # that exact key ring when verifying a user's signature. That way we can make
+  # sure one user cannot just upload multiple signatures to reach the 5/7
+  # required sigs.
+  for key in "${KEYS[@]}"; do
+    KEY_ID=$(echo $key | cut -d' ' -f1)
+    IMPORT_FILE=$(echo $key | cut -d' ' -f2)
+    KEY_FILE="$DIR/$IMPORT_FILE"
+    BASENAME=$(basename "$KEY_FILE")
+    USERNAME=${BASENAME%.*}
+    KEYRING="$TEMP_DIR/$USERNAME.pgp"
+
+    # We'll write the status of the import to a special file that we can then
+    # inspect.
+    STATUS_FILE="$TEMP_DIR/$USERNAME.import-status"
+
+    echo "Importing key for user $USERNAME from $key"
+    gpg --no-default-keyring --keyring "$KEYRING" --import \
+      --status-fd=1 < "$KEY_FILE" > "$STATUS_FILE" 2>&1
+
+    # Make sure we only imported exactly one key.
+    if ! grep -q "imported: 1\$" "$STATUS_FILE"; then
```

Comment:
This can be bypassed:

1. `gpg --quick-generate-key 'imported: 1 <alice@example.com>'`
2. `gpg --export 'imported: 1' >> malicious_key.gpg`

Did I tell you I hate text interfaces? :)

Reply:
Actually, it wouldn't since the dollar sign matches the line end. But yeah, text interfaces are ugly.
I ended up replicating the behavior of `sqck`, it does seem to be the most fool proof way. Nice tool by the way!

Reply:
Ah, yes, I missed the dollar sign, sorry. Went to check what happens if one puts `\n` into key name - it wouldn't fool this check as the output is escaped although some other outputs are not escaped.

Will do the rest of review a bit later.